src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -617,24 +703,30 @@ public MedicalInventory actualizeMedicalInventoryRow(MedicalInventory inventory)
 			}
 		}
 		// Get all the lot of the movements
-		List<Lot> lotOfMovements = movs.stream().map(Movement::getLot).collect(Collectors.toList());
+		List<Lot> lotsFromMovements = movs.stream().map(Movement::getLot).collect(Collectors.toList());
+		// Get latest movement date
+		Movement latestMovement = movBrowserManager.getLastMovement();
+		LocalDateTime latestMovementDate = null;
+		if (latestMovement != null) {
+			latestMovementDate = latestMovement.getDate();
+		}
 		// Remove duplicates by converting the list to a set
-		Set<Lot> uniqueLots = new HashSet<>(lotOfMovements);
+		Set<Lot> uniqueLots = new HashSet<>(lotsFromMovements);
 		// Convert the set back to a list
 		List<Lot> uniqueLotList = new ArrayList<>(uniqueLots);
 		// Cycle fetched movements to see if they impact inventoryRowSearchList
 		for (Lot lot : uniqueLotList) {
-			String lotCodeOfMovement = lot.getCode();
+			String lotCode = lot.getCode();
 			Medical medical = lot.getMedical();
 			Integer medicalCode = medical.getCode();
 			// Fetch also empty lots because some movements may have discharged them completely
-			Optional<Lot> optLot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCodeOfMovement))
+			Optional<Lot> optLot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCode))
 				.findFirst();
 			double mainStoreQty = optLot.get().getMainStoreQuantity();
```

Comment:
Should there be a `optLot.isPresent()` or `!optLot.isEmpty()` call before accessing `optLot` here?  Line 723 says the value is `Optional`

Reply:
Yes... fixed, thank!